### ISSUE TITLE
Error: Attribute "mainColor" has already been defined

### ISSUE THREAD

**Issue by nex0s** [Mapwize/mapwize-android-sdk#5]
This error is due a conflict with another library we are using in our project.

The problem is that the same attribute name is used more than once, which is not allowed by the Android SDK build tools, which is due to bad design: https://code.google.com/p/android/issues/detail?id=22576

MapWize defines a custom attribute with the name 'mainColor' for the MWZMapView custom view at https://github.com/Mapwize/mapwize-android-sdk/blob/10409f993f85ab0730c94e08c458d2b7dac0fcfd/mapwize/src/main/res/values/attrs.xml#L14

<resources>
    <declare-styleable name="MWZMapView">
        <attr name="accesskey" format="string" />
        <attr name="language" format="string" />
        <attr name="showUserPositionControl" format="boolean" />
        <attr name="displayFloorControl" format="boolean" />
        <attr name="mainColor" format="string" />
        <attr name="isLocationEnabled" format="boolean" />
        <attr name="isBeaconsEnabled" format="boolean" />
        <attr name="center_latitude" format="float" />
        <attr name="center_longitude" format="float" />
        <attr name="zoom" format="integer" />
        <attr name="floor" format="integer" />
        <attr name="minZoom" format="integer" />
        <attr name="maxBounds_latitudeMin" format="float" />
        <attr name="maxBounds_latitudeMax" format="float" />
        <attr name="maxBounds_longitudeMin" format="float" />
        <attr name="maxBounds_longitudeMax" format="float" />
        <attr name="bounds_latitudeMin" format="float" />
        <attr name="bounds_latitudeMax" format="float" />
        <attr name="bounds_longitudeMin" format="float" />
        <attr name="bounds_longitudeMax" format="float" />
        <attr name="iconUrl" format="string" />
        <attr name="x_iconSize" format="integer" />
        <attr name="y_iconSize" format="integer" />
        <attr name="x_iconAnchor" format="integer" />
        <attr name="y_iconAnchor" format="integer" />
    </declare-styleable>
</resources>

We recommend that you put a prefix to all library resources so that conflicts can be avoided.

tl;dr

attribute / resource IDs are global, even across libraries
use prefix when declaring attributes / resources.
To fix this problem MWZMapView definition in style.xml needs to change to something like this:

<resources>
    <declare-styleable name="MWZMapView">
        <attr name="mwz_accesskey" format="string" />
        <attr name="mwz_language" format="string" />
        <attr name="mwz_showUserPositionControl" format="boolean" />
        <attr name="mwz_displayFloorControl" format="boolean" />
        <attr name="mwz_mainColor" format="string" />
        <attr name="mwz_isLocationEnabled" format="boolean" />
        <attr name="mwz_isBeaconsEnabled" format="boolean" />
        <attr name="mwz_center_latitude" format="float" />
        <attr name="mwz_center_longitude" format="float" />
        <attr name="mwz_zoom" format="integer" />
        <attr name="mwz_floor" format="integer" />
        <attr name="mwz_minZoom" format="integer" />
        <attr name="mwz_maxBounds_latitudeMin" format="float" />
        <attr name="mwz_maxBounds_latitudeMax" format="float" />
        <attr name="mwz_maxBounds_longitudeMin" format="float" />
        <attr name="mwz_maxBounds_longitudeMax" format="float" />
        <attr name="mwz_bounds_latitudeMin" format="float" />
        <attr name="mwz_bounds_latitudeMax" format="float" />
        <attr name="mwz_bounds_longitudeMin" format="float" />
        <attr name="mwz_bounds_longitudeMax" format="float" />
        <attr name="mwz_iconUrl" format="string" />
        <attr name="mwz_x_iconSize" format="integer" />
        <attr name="mwz_y_iconSize" format="integer" />
        <attr name="mwz_x_iconAnchor" format="integer" />
        <attr name="mwz_y_iconAnchor" format="integer" />
    </declare-styleable>
</resources>

**Comment by mathieugerard** [comment]
Hi nex0s,
Thanks for the feedback. We are looking into it.
Regards
Mathieu

**Comment by mathieugerard** [comment]
Hi nex0s,
This SDK has been deprecated for a long time now. We recommend you to migrate to the newest MapwizeForMapbox if you didn't do that yet. Have a look at docs.mapwize.io.
Regards
Mathieu
